pkg/registry/core/pod/strategy.go

```diff
@@ -371,6 +382,12 @@ func podHasContainerWithName(pod *api.Pod, containerName string) bool {
 			return true
 		}
 	}
+	for _, ec := range pod.Spec.EphemeralContainers {
```

Comment:
This sort of check is really common, have you verified all the places in the codebase also have it?

Reply:
I added a commit (currently https://github.com/kubernetes/kubernetes/pull/59416/commits/edaa7099c53983d5b4c8ff7bce70076421c8e22d) with a new helper for iterating containers in a pod spec which should make this sort of thing easier.

I haven't replaced all occurrences yet, but please chime in if you don't like its look.